pkg/apis/kudo/v1beta1/operatorversion_types.go

```diff
@@ -109,6 +109,9 @@ type Parameter struct {
 
 	// Type specifies the value type. Defaults to `string`.
 	Type ParameterType `json:"value-type,omitempty"`
+
+	// Specifies if the parameter can be changed after the initial installation of the operator
+	Immutable *bool `json:"immutable,omitempty"`
```

Comment:
Why is `Immutable` a bool ptr? The [immutable -> mutable](https://github.com/kudobuilder/kudo/pull/1575/files#diff-c10c73f4220d3a0c71475b7002482b65R387) transition is allowed and doesn't require a validation. So why can't we keep it a `bool` and treat all parameters as `immutable = false` by default?

Reply:
Tbh, I'm not sure. I copied it from the `required` definition. I was wondering why "required" is a pointer though.

I guess it renders a little cleaner yaml if the bools are not set? But I don't really mind either way.